tests/core/CoreTest.py

```diff
@@ -383,7 +383,8 @@ def execute_run(bears):
         def on_result(result):
             results.append(result)
 
-        run(bears, on_result)
+        session = Session(bears, on_result)
+        session.run()
```

Comment:
I would rather add an extra parameter in `execute_run()` function related to `on_result`. So, that it can be reused instead of having different functions.